### DISCUSSION TITLE
Can somebody share the /root/.bashrc?

### DISCUSSION THREAD

**Discussion by norbertjoni** [wulfy23/rpi4#33] [Q&A] [answered]
Hi guys,

Could you share your /root/.bashrc file with me, please? I don't use wulfy23's built at the moment but I would like a nice bash for my rpi.

Thank you.

**Comment by wulfy23** [comment, marked as answer]
howdy,

well kind of long answer but in terms of bashrc etc. 

1. there are quite a few levels / components that make up what get's spat out and checked during login
2. the places (some of) those bits are placed are a little convoluted for someone to replicate all

maybe if you specify exactly which areas you are most interested in I would be better able to advise / know what to provide you with (which i'd be happy to do)

i.e.)

1. login "sysinfo" (see README for original coder / attribution) - ip info, disk space info mostly
(edit might not be OP repo but a reference https://github.com/Rafciq/openwrt/blob/master/misc/sysinfo.sh )


3. connected client / "dhcp/arp/other seen" printout
4. special warnings i.e. you have been (rpi) power throttled etc
5. the prompt itself (this is implemented pretty "overkill"-ish due to ash and bash etc. etc. basically we don't assume bash so pretty much all of my stuff must be conditional and support both (or panic obviously etc.)
6. custom messaging wrappers (splitting / duplicating / routing script messages to console/terminal/syslog/kmsg)
7. um.... fsck filesystem warnings and various other stuff might also hook in (semi persistent state dotfiles)
8. calling of external features and routines (i.e. "lucisshon" )


my guess is I can probably point you in the right direction for 1 some of 4 with perhaps some samples then advise on what you are looking to do from there a little ( from that point 95% of stuff is pretty basic bash/ash that is not so readable in the format it's used within the build... )

just don't wanna slam up 12000 lines of code half of it random comments / layman trial and error where you probably only need around 100 lines maximim.... ( having said that there should be a "rootfs" somewhere of not I will put one up and point you to the relevant files if you wish )

**Reply by norbertjoni** [reply]
Wow, thank you.  Meantime I saw there is an update in your repo. Firstly I will try to install it and will see how happy am I :) 

One question: there is any tweak for shutdown the rpi peacefully in your build? I am thinking about sd card corruption. I had a read-only issue in the official build.

**Reply by wulfy23** [reply]
The build handles ext4 (and fat) fscking for you and will repair itself.
(aka it is a BOOTUP / rc.local-ish (technically pre-init) level workaround / fix) > "if fsck fails then fsck and reboot"

Regarding shutting down, it also tries to be a little more graceful maybe... ( but all the bells an whistles mean it possibly does not )

"shutting down" is an interesting topic... alot of it is "pi foundation / rapberry pi limited" or whatever the name is now. ( aka it is a weird mix of hardware / bootloader and os....

what I do... is issue a "reboot" wait for around 12-15 seconds until a board led goes off then unplug / hard switch off... but yeah... it is kinda tricky knowing what is "graceful" / best practice on these...

**Comment by norbertjoni** [comment]
I installed your build with my backup and all looks okay except my terminal:

![image](https://user-images.githubusercontent.com/25225562/215808887-41acd235-4108-47ac-8e7d-c440123c1cb7.png)
How can I fix it?

**Reply by norbertjoni** [reply]
nevermind, I will reinstall and restore with the necessary files only.
